src/control/server/ioserver/exec.go

```diff
@@ -139,16 +146,23 @@ func (r *Runner) Start(ctx context.Context, errOut chan<- error) error {
 	return nil
 }
 
-func (r *Runner) setStarted() {
-	atomic.StoreUint32(&r.started, 1)
+func (r *Runner) setRunning() {
+	atomic.StoreUint32(&r.running, 1)
 }
 
 func (r *Runner) setStopped() {
-	atomic.StoreUint32(&r.started, 0)
+	atomic.StoreUint32(&r.running, 0)
 }
 
-func (r *Runner) IsStarted() bool {
-	return atomic.LoadUint32(&r.started) == 1
+func (r *Runner) IsRunning() bool {
+	return atomic.LoadUint32(&r.running) == 1
+}
+
+func (r *Runner) Stop() error {
+	if !r.IsRunning() {
+		return errors.Errorf("instance %d is not running", r.Config.Index)
+	}
```

Comment:
Not sure that this is a useful error, tbh. What should the caller do with it? It's still worthwhile to check the running state to avoid trying to kill a process that's already killed, but IMO it should just return nil here to make Stop() idempotent.